products/dialogflow/api.yaml

```diff
@@ -0,0 +1,123 @@
+         from the API will be shown in the [avatarUriBackend] field.
+      - !ruby/object:Api::Type::String
+        name: 'avatarUriBackend'
+        description: |
+         The URI of the agent's avatar as returned from the API. Output only. To provide an image URL for the agent avatar,
+         the [avatarUri] field can be used.
+        output: true
+      - !ruby/object:Api::Type::Boolean
+        name: 'enableLogging'
+        description: |
+         Determines whether this agent should log conversation queries.
+      - !ruby/object:Api::Type::Enum
+        name: 'matchMode'
+        description: |
+          Determines how intents are detected from user queries.
+          * MATCH_MODE_HYBRID: Best for agents with a small number of examples in intents and/or wide use of templates syntax and composite entities.
+          * MATCH_MODE_ML_ONLY: Can be used for agents with a large number of examples in intents, especially the ones using @sys.any or very large developer entities.
+        values:
+          - :MATCH_MODE_HYBRID
+          - :MATCH_MODE_ML_ONLY
+      - !ruby/object:Api::Type::Double
+        name: 'classificationThreshold'
+        description: |
+         To filter out false positive results and still get variety in matched natural language inputs for your agent, you can tune the machine learning classification threshold. 
+         If the returned score value is less than the threshold value, then a fallback intent will be triggered or, if there are no fallback intents defined, no intent will be 
+         triggered. The score values range from 0.0 (completely uncertain) to 1.0 (completely certain). If set to 0.0, the default of 0.3 is used.
+      - !ruby/object:Api::Type::Enum
+        name: 'apiVersion'
+        description: |
+          API version displayed in Dialogflow console. If not specified, V2 API is assumed. Clients are free to query different service endpoints for different API versions. 
+          However, bots connectors and webhook calls will follow the specified API version.
+          * API_VERSION_V1: Legacy V1 API.
+          * API_VERSION_V2: V2 API.
+          * API_VERSION_V2_BETA_1: V2beta1 API.
+        values:
+          - :API_VERSION_V1
+          - :API_VERSION_V2
+          - :API_VERSION_V2_BETA_1
+      - !ruby/object:Api::Type::Enum
+        name: 'tier'
+        description: |
+          The agent tier. If not specified, TIER_STANDARD is assumed.
+          * TIER_STANDARD: Standard tier.
+          * TIER_ENTERPRISE: Enterprise tier (Essentials).
+          * TIER_ENTERPRISE_PLUS: Enterprise tier (Plus).
+          NOTE: This field seems to have eventual consistency in the API. Updating this field to a new value, or even creating a new agent with a tier that is different from a previous agent 
+          in the same project will take some time to propogate. The provider will wait for the API to show consistency, which can lead to longer apply times.
```

Comment:
Downstream travis builds are failing because:
> google-beta/resource_dialogflow_agent.go:126:44: `propogate` is a misspelling of `propagate` (misspell)